assets/js/components/IngestSheet/List.jsx

```diff
@@ -68,46 +68,49 @@ const IngestSheetList = ({ project, subscribeToIngestSheetStatusChanges }) => {
               </tr>
             </thead>
             <tbody>
-              {project.ingestSheets.map(({ id, title, status, updatedAt }) => (
-                <tr key={id}>
-                  <td>
-                    <Link to={`/project/${project.id}/ingest-sheet/${id}`}>
-                      {title}
-                    </Link>
-                  </td>
-                  <td className="has-text-right">{formatDate(updatedAt)}</td>
-                  <td>
-                    <span
-                      className={`tag ${getClassFromIngestSheetStatus(status)}`}
-                    >
-                      {TEMP_USER_FRIENDLY_STATUS[status]}
-                    </span>
-                  </td>
-                  <td className="has-text-right">
-                    {["APPROVED", "COMPLETED"].indexOf(status) > -1 && (
-                      <Link
-                        to={`/project/${project.id}/ingest-sheet/${id}`}
-                        className="button"
-                      >
-                        {<FontAwesomeIcon icon="eye" />}{" "}
-                        <span className="sr-only">View</span>
+              {project.ingestSheets.map(
+                ({ id, title, status, updatedAt }, index) => (
+                  <tr key={id + index}>
```

Comment:
Is `id` not unique here?  How does adding the `index` value help?

Reply:
The actual bug is that there is a console error for duplicity of keys and then the rows wouldn't update once the error occurs. Id is not unique due to changing status messages (like validation pending/validated etc where id is same but status is not). Adding index provides uniqueness so there are no errors and the rows update when state changes.